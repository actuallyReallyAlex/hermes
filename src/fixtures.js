export const defaultState = {
  ship: {
    cargo: {
      items: [],
      volumeRemaining: 5
    },
    location: {
      name: null,
      value: null
    }
  },
  ui: {
    view: 'Ship'
  },
  user: {
    cash: 100
  },
  world: {
    isTimerRunning: false,
    planets: [
      {
        id: "0",
        isHomePlanet: true,
        location: 0,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '0',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '1',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '2',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '3',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '4',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 1'
      },
      {
        id: "1",
        isHomePlanet: false,
        location: 50,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '5',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '6',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '7',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '8',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '9',
            destination: {
              name: 'Test Planet 1',
              value: 0
            }
          }
        ],
        name: 'Test Planet 2'
      },
      {
        id: "2",
        isHomePlanet: false,
        location: 100,
        items: [
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '10',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '11',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '12',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '13',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          },
          {
            name: 'Test Item',
            space: 1,
            value: 1,
            id: '14',
            destination: {
              name: 'Test Planet 2',
              value: 50
            }
          }
        ],
        name: 'Test Planet 3'
      }
    ]
  }
}
