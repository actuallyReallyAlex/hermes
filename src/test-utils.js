import React from 'react'
import configureStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { defaultState } from './fixtures'
import { render } from '@testing-library/react'

const mockThunks = ({ dispatch, getState }) => next => action => {
  const mockedAction = () => ({ type: 'MOCKED ACTION' })

  let returnValue

  if (typeof action === 'function' || action === undefined) {
    returnValue = mockedAction
  } else {
    returnValue = next(action)
  }

  return returnValue
}

const middlewares = [mockThunks]

export const mockStore = configureStore(middlewares)

export const customRender = options => {
  const { component, state, props } = options

  const Component = component

  return render(
    <Provider store={mockStore(state ? state : defaultState)}>
      <Component {...props} />
    </Provider>
  )
}
