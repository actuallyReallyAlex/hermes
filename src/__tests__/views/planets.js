import { customRender } from '../../test-utils'
import { defaultState } from '../../fixtures'
import PlanetsView from '../../views/planets'
import { waitForElement, fireEvent } from '@testing-library/dom'

const customState = {
  ...defaultState,
  ship: {
    ...defaultState.ship,
    location: { name: 'Test Planet 1', value: 0 },
    cargo: {
      items: [
        {
          destination: {
            name: 'Test Planet 2',
            value: 50
          },
          id: '0',
          name: 'Test Item',
          quantity: 1,
          value: 1,
          volume: 1
        }
      ],
      volumeRemaining: 4
    }
  }
}

describe('<PlanetsView />', () => {
  it('Should render the <PlanetsView /> component.', () => {
    const container = customRender({
      component: PlanetsView,
      state: customState
    })
    expect(container.asFragment()).toMatchSnapshot()
  })

  it('Should handle traveling.', async () => {
    const { getByTestId } = customRender({
      component: PlanetsView,
      state: customState
    })
    const button = await waitForElement(() =>
      getByTestId('travel-button-Test Planet 2')
    )
    fireEvent.click(button)
  })

  it('Should handle adding an item to ship cargo.', async () => {
    const { getByTestId } = customRender({
      component: PlanetsView,
      state: customState
    })
    const button = await waitForElement(() => getByTestId('add-button-1'))
    fireEvent.click(button)
  })
})
