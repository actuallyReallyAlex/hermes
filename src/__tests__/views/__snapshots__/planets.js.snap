// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlanetsView /> Should render the <PlanetsView /> component. 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 gQLnBt"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 yhirM"
        >
          Test Planet 1 - Home Planet
        </h2>
      </div>
      <span>
        Items:
      </span>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <button
            class="StyledButton-sc-323bzc-0 jJyttA"
            data-testid="add-button-0"
            disabled=""
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 iOkQrb"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <label
            for="quantity-input-0"
          >
            Quantity to Add
          </label>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <input
            id="quantity-input-0"
            max="4"
            min="0"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <button
            class="StyledButton-sc-323bzc-0 jJyttA"
            data-testid="add-button-1"
            disabled=""
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 iOkQrb"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <label
            for="quantity-input-1"
          >
            Quantity to Add
          </label>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <input
            id="quantity-input-1"
            max="4"
            min="0"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <button
            class="StyledButton-sc-323bzc-0 jJyttA"
            data-testid="add-button-2"
            disabled=""
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 iOkQrb"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <label
            for="quantity-input-2"
          >
            Quantity to Add
          </label>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <input
            id="quantity-input-2"
            max="4"
            min="0"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <button
            class="StyledButton-sc-323bzc-0 jJyttA"
            data-testid="add-button-3"
            disabled=""
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 iOkQrb"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <label
            for="quantity-input-3"
          >
            Quantity to Add
          </label>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <input
            id="quantity-input-3"
            max="4"
            min="0"
            type="number"
            value="0"
          />
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <button
            class="StyledButton-sc-323bzc-0 jJyttA"
            data-testid="add-button-4"
            disabled=""
            type="button"
          >
            <svg
              aria-label="Add"
              class="StyledIcon-ofa7kd-0 iOkQrb"
              viewBox="0 0 24 24"
            >
              <path
                d="M12,22 L12,2 M2,12 L22,12"
                fill="none"
                stroke="#000"
                stroke-width="2"
              />
            </svg>
          </button>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <label
            for="quantity-input-4"
          >
            Quantity to Add
          </label>
          <div
            class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hPsznL"
          />
          <input
            id="quantity-input-4"
            max="4"
            min="0"
            type="number"
            value="0"
          />
        </div>
      </div>
    </div>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 gQLnBt"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 yhirM"
        >
          Test Planet 2
        </h2>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <button
          class="StyledButton-sc-323bzc-0 jAPRVi"
          data-testid="travel-button-Test Planet 2"
          type="button"
        >
          <svg
            aria-label="Target"
            class="StyledIcon-ofa7kd-0 iOkQrb"
            viewBox="0 0 24 24"
          >
            <path
              d="M23,12 C23,18.075 18.075,23 12,23 C5.925,23 1,18.075 1,12 C1,5.925 5.925,1 12,1 C18.075,1 23,5.925 23,12 L23,12 Z M18,12 C18,8.691 15.309,6 12,6 C8.691,6 6,8.691 6,12 C6,15.309 8.691,18 12,18 C15.309,18 18,15.309 18,12 L18,12 Z M13,12 C13,11.448 12.552,11 12,11 C11.448,11 11,11.448 11,12 C11,12.552 11.448,13 12,13 C12.552,13 13,12.552 13,12 L13,12 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <span
          class="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          ETA: 8 minutes 20 seconds
        </span>
      </div>
      <span>
        Items:
      </span>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 1
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 1
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 1
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 1
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 1
          </span>
        </div>
      </div>
    </div>
    <div>
      <div
        class="StyledBox-sc-13pk1d4-0 gQLnBt"
      >
        <h2
          class="StyledHeading-sc-1rdh4aw-0 yhirM"
        >
          Test Planet 3
        </h2>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <button
          class="StyledButton-sc-323bzc-0 jAPRVi"
          data-testid="travel-button-Test Planet 3"
          type="button"
        >
          <svg
            aria-label="Target"
            class="StyledIcon-ofa7kd-0 iOkQrb"
            viewBox="0 0 24 24"
          >
            <path
              d="M23,12 C23,18.075 18.075,23 12,23 C5.925,23 1,18.075 1,12 C1,5.925 5.925,1 12,1 C18.075,1 23,5.925 23,12 L23,12 Z M18,12 C18,8.691 15.309,6 12,6 C8.691,6 6,8.691 6,12 C6,15.309 8.691,18 12,18 C15.309,18 18,15.309 18,12 L18,12 Z M13,12 C13,11.448 12.552,11 12,11 C11.448,11 11,11.448 11,12 C11,12.552 11.448,13 12,13 C12.552,13 13,12.552 13,12 L13,12 Z"
              fill="none"
              stroke="#000"
              stroke-width="2"
            />
          </svg>
        </button>
        <div
          class="StyledBox__StyledBoxGap-sc-13pk1d4-1 hyyRSo"
        />
        <span
          class="StyledText-sc-1sadyjn-0 fWSbXS"
        >
          ETA: 16 minutes 40 seconds
        </span>
      </div>
      <span>
        Items:
      </span>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
      </div>
      <div
        class="StyledBox-sc-13pk1d4-0 xqboQ"
      >
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Name: Test Item
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Quantity: 
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Volume: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Value: 1
          </span>
        </div>
        <div
          class="StyledBox-sc-13pk1d4-0 hroMnY"
        >
          <span
            class="StyledText-sc-1sadyjn-0 fWSbXS"
          >
            Destination: Test Planet 2
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
